components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/internal/ClientManager.java

```diff
@@ -51,86 +67,155 @@ public class ClientManager {
-        // Increase max total connection to 20.
-        poolingHttpClientConnectionMgr.setMaxTotal(maxConnections);
-        // Increase default max connection per route to 20.
-        poolingHttpClientConnectionMgr.setDefaultMaxPerRoute(maxConnectionsPerRoute);
-        return poolingHttpClientConnectionMgr;
+        TlsStrategy tlsStrategy = ClientTlsStrategyBuilder.create()
+                .setSslContext(getSSLContext())
+                .setHostnameVerifier(new DefaultHostnameVerifier())
+                .build();
+
+        return PoolingAsyncClientConnectionManagerBuilder.create()
+                .setTlsStrategy(tlsStrategy)
+                .setPoolConcurrencyPolicy(PoolConcurrencyPolicy.STRICT)
+                .setConnPoolPolicy(PoolReusePolicy.LIFO)
+                .setMaxConnTotal(maxConnections)
+                .setMaxConnPerRoute(maxConnectionsPerRoute)
+                .setDefaultConnectionConfig(createConnectionConfig())
+                .setDefaultTlsConfig(createTlsConfig())
+                .build();
     }
 
-    private void addSslContext(HttpAsyncClientBuilder builder) throws WebSubAdapterException {
+    private HttpClientConnectionManager createPoolingConnectionManager() throws WebSubAdapterException {
 
-        try {
-            SSLContext sslContext = SSLContexts.custom()
-                    .loadTrustMaterial(WebSubHubAdapterDataHolder.getInstance().getTrustStore(), null)
-                    .build();
-            builder.setSSLContext(sslContext);
-            builder.setSSLHostnameVerifier(new DefaultHostnameVerifier());
-
-        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
-            throw handleServerException(ERROR_CREATING_SSL_CONTEXT, e);
+        int maxConnections = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnections();
+        int maxConnectionsPerRoute = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnectionsPerRoute();
+
+        SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(
+                getSSLContext(), new DefaultHostnameVerifier());
+
+        return PoolingHttpClientConnectionManagerBuilder.create()
+                .setSSLSocketFactory(sslConnectionSocketFactory)
+                .setPoolConcurrencyPolicy(PoolConcurrencyPolicy.STRICT)
+                .setConnPoolPolicy(PoolReusePolicy.LIFO)
+                .setMaxConnTotal(maxConnections)
+                .setMaxConnPerRoute(maxConnectionsPerRoute)
+                .setDefaultConnectionConfig(createConnectionConfig())
+                .setDefaultTlsConfig(createTlsConfig())
+                .build();
+    }
+
+    private SSLContext getSSLContext() throws WebSubAdapterServerException {
+
+        if (sslContext == null) {
+            LOG.info("Creating SSL context for WebSub outbound adapter.");
```

Comment:
This log would seem out of place as the others are debug logs.

Reply:
Resolved